### PR TITLE
Changed FindFlyCapture2.cmake so it works in Windows as well

### PR DESCRIPTION
**Pull request by VileLasagna**
The initial form of this saved me quite some pain. Thought I'd give back my own improvements on it.

### CODE REVIEW

**Comment by kartikmohta** [comment]
@VileLasagna You're using the ROS driver in Windows?

**Comment by VileLasagna** [comment]
I'm not really using your drivers. I just needed to use Flycap in a different project and found your CMake script. So this is basically all I know from this repo

**Comment by versatran01** [comment]
@kartikmohta You can merge this if you think this is ok.

File: cmake/FindFlyCapture2.cmake [hunk 1]
```diff
@@ -71,11 +79,31 @@ list(APPEND FlyCapture2_CHECK_INCLUDE_DIRS
 list(APPEND FlyCapture2_CHECK_LIBRARY_DIRS
     ${FLYCAPTURE_DIR}/flycapture/lib
     )
+if(MSVC)
```
Comment (kartikmohta) [inline cmake/FindFlyCapture2.cmake]: Shouldn't this be `if(WIN32)`?

File: cmake/FindFlyCapture2.cmake [hunk 2]
```diff
@@ -84,67 +112,162 @@ set(FlyCapture2_FOUND TRUE)
...
 endif()
 
 # Catch case when caller has set FlyCapture2_INCLUDE_DIR in the cache / GUI and
 # thus FIND_[PATH/LIBRARY] are not called, but specified locations are
 # invalid, otherwise we would report the library as found.
-if(FlyCapture2_INCLUDE_DIR AND NOT EXISTS ${FlyCapture2_INCLUDE_DIR}/flycapture/FlyCapture2.h)
-    FlyCapture2_REPORT_NOT_FOUND("Caller defined FlyCapture2_INCLUDE_DIR: "
-        ${FlyCapture2_INCLUDE_DIR}
-        " does not contain flycapture/FlyCapture22.h header.")
+if(MSVC)
```
Comment (kartikmohta) [inline cmake/FindFlyCapture2.cmake]: `if(WIN32)` here?

File: cmake/FindFlyCapture2.cmake [hunk 3]
```diff
@@ -84,67 +112,162 @@ set(FlyCapture2_FOUND TRUE)
...
+if(NOT WIN32)
 # Find library directory for flycapture
 find_library(FlyCapture2_LIBRARY
     NAMES libflycapture.so
     PATHS ${FlyCapture2_LIBRARY_DIR_HINTS}
     ${FlyCapture2_CHECK_LIBRARY_DIRS}
     )
+else()
+
+    if(MSVC)
```
Comment (kartikmohta) [inline cmake/FindFlyCapture2.cmake:153]: I think the `if(MSVC)` is only needed for the following block for setting `vTail` but not needed for the later `find_library` and `find_file`.

File: cmake/FindFlyCapture2.cmake [hunk 4]
```diff
@@ -84,67 +112,162 @@ set(FlyCapture2_FOUND TRUE)
...
+            ${FlyCapture2_INCLUDE_DIR}
+            " does not contain flycapture/FlyCapture2.h header.")
+    endif()
 endif()
 
 # Set standard CMake FindPackage variables if found.
 if(FlyCapture2_FOUND)
     set(FlyCapture2_INCLUDE_DIRS ${FlyCapture2_INCLUDE_DIR})
-    file(GLOB FlyCapture2_LIBRARIES ${FlyCapture2_LIBRARY})
+    if(MSVC)
```
Comment (kartikmohta) [inline cmake/FindFlyCapture2.cmake]: `if(WIN32)`?

**Review by kartikmohta — COMMENTED** [review]
@VileLasagna Have a few comments about the use of `if(MSVC)` vs `if(WIN32)`. Let me know if you agree with the proposed changes.

**Comment by VileLasagna** [comment]
Hey

The reason I've hidden these behind MSVC checks is because of CYGWIN and MINGW. They're both windows platforms but I'm honestly not sure how they'd link to these, if you were using one of those environments (something like clang-cl still counts as MSVC because of vcvarsall.bat).

My home machines are running only Linux and at work it's all VC so I don't have the environment properly set up to quickly test those so it's more me not remembering how this goes once you go down that path. A quick googling says you're probably OK if you're linking with C idiom so... I can check Flycap tomorrow and if it's C idiom default to the vc140 version, toss a warning and hope for the best?

**Comment by kartikmohta** [comment]
The places I've marked are not dependent on the compiler being used (for example, when finding the location of header files or lib files) so hopefully they should be fine with MSVC or MINGW. Cygwin most likely won't work at all for this.

**Comment by VileLasagna** [comment]
I still think it shouldn't link but if you think it's a better option I can have it point to the newest library, throw a warning and hope for the  best?

**Comment by kartikmohta** [comment]
I'm not sure either whether it will link, but for now I suppose you can point to the newest one and we'll hope that somebody comes along, tests and reports a bug if it doesn't work 🙂

**Comment by VileLasagna** [comment]
I've implemented the changes we talked about. 
As part of the warning message I've suggested the user look into the C Idiom libraries Point Grey provides as part of their FlyCapture package. Since they're different libraries, however, I haven't adapted the script to actually give you those though, but they should be a more hopeful option for people trying to link FlyCap while building with CYGWIN or MINGW

**Comment by kartikmohta** [comment]
Thanks @VileLasagna

**Pull request merged**